### PR TITLE
[release-4.17] OCPBUGS-39204: V1.36.1 rebase release 4.17

### PR DESCRIPTION
**Pull request by arkadeepsen**
Manual cherry-pick of #387

Followed the below steps:

- Rebase to upstream v1.36.1

  - Ran the following script to create the rebase branch:

    ```
    ./UPSTREAM-MERGE.sh v1.36.1 release-4.17
    ```
Manually cherry-picked the following commits
- Regenerate 03-setversion.patch (a87ee6cb5139261a8b62816470fbafa9ba233264)
- Regenerate fixsanity patch (abd16bde54e64f0eae1b07ad00d8264c09777d25)
- Regenerate disable security context patch (2700f259801621e32b34f251bc6c7f6ef0df18a0)
- Regenerate 06-build-darwin-binary.patch (5c5075c1591095371940ca8c1ebc8e72dfd363f0)
- Regenerate 11-openshiftv1-cli.patch (2e1f7158a99442170becacd1e8ebef97d26c47de)
- Remove ansible code (a3930557b5cc34970399591997a3b175128f61a9)
- Update vendor directory (33361b4a230a2d688e1fa39f007e2efdca47a4c4)
- Set the value of GOBIN in e2e-helm.sh before "make install" of operator-sdk (94c408f056a4418e381839384648097f887e91f4)
- Update OCP version (5949f3ba4ec1f11f28173f3f2ef214e6b14dd982) and modified OCP version to 4.17
- Update UBI minimal version (288e744fc419c3ef3c0aa977b886c35863f83370)
- Remove unused symlink of Dockerfile (615e06e935e43e198339eb2a2073975d12a41f5c)

### CODE REVIEW

File: internal/plugins/openshift/v1/init.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ import (
 
 const (
 	// The current OCP release version.
-	ocpProductVersion = "4.15"
+	ocpProductVersion = "4.17"
```
Comment (chiragkyal) [inline internal/plugins/openshift/v1/init.go:30]: Apart from this change, I hope everything else is exact same as https://github.com/openshift/ocp-release-operator-sdk/pull/387

**Review by KeenonLee — COMMENTED** [review]
/label cherry-pick-approved

File: internal/plugins/openshift/v1/init.go (hunk 1 above)
Comment (arkadeepsen) [reply internal/plugins/openshift/v1/init.go:30]: Yes. That's correct.

**Comment by arkadeepsen** [comment]
/hold
There's some issue with release image for 4.18: https://redhat-internal.slack.com/archives/CB95J6R4N/p1725000287781299

**Comment by arkadeepsen** [comment]
/hold cancel
https://github.com/openshift/ocp-release-operator-sdk/pull/394 was used to successfully build operator-sdk and helm-operator images. Ref: https://github.com/openshift/ocp-release-operator-sdk/pull/394#issuecomment-2338487115 & https://github.com/openshift/ocp-release-operator-sdk/pull/394#issuecomment-2338494097.

**Pull request merged**